### PR TITLE
Add refresh cache logic to keep docker and sfx in sync

### PR DESCRIPTION
**Pull request by dloucasfx**
Add refresh cache logic to keep docker and sfx in sync when the event stream fails

cc: @keitwb @jrcamp 
Signed-off-by: Dani Louca <dlouca@splunk.com>

### CODE REVIEW

File: pkg/core/common/docker/containerlist.go [hunk 1]
```diff
@@ -117,33 +172,9 @@ func ListAndWatchContainers(ctx context.Context, client *docker.Client, changeHa
 
 	<-watchStarted
 
-	f := filters.NewArgs()
-	f.Add("status", "running")
-	options := dtypes.ContainerListOptions{
-		Filters: f,
-	}
-	containerList, err := client.ContainerList(ctx, options)
+	err := syncContainerList()
```
Comment (keitwb) [inline pkg/core/common/docker/containerlist.go:175]: Move this to immediately before the for loop on line 122 and then always return nil err from this method.  We don't want the monitor to crash and not work if docker happens to be down when the agent starts up.

Comment (keitwb) [reply pkg/core/common/docker/containerlist.go:175]: Then you can get rid of the watchStarted stuff as well.

Comment (dloucasfx) [reply pkg/core/common/docker/containerlist.go]: done. 
also did more cleanup for unused

**Review by keitwb — APPROVED** [review]
I can't see any edge cases that aren't covered.  This should be quite a bit better.  Thanks for adding this.

**Pull request merged**